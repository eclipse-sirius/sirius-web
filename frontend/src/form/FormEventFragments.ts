/*******************************************************************************
 * Copyright (c) 2021 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import gql from 'graphql-tag';

export const subscribersUpdatedEventPayloadFragment = gql`
  fragment subscribersUpdatedEventPayloadFragment on SubscribersUpdatedEventPayload {
    id
    subscribers {
      username
    }
  }
`;

export const widgetSubscriptionsUpdatedEventPayloadFragment = gql`
  fragment widgetSubscriptionsUpdatedEventPayloadFragment on WidgetSubscriptionsUpdatedEventPayload {
    id
    widgetSubscriptions {
      widgetId
      subscribers {
        username
      }
    }
  }
`;

export const formRefreshedEventPayloadFragment = gql`
  fragment formRefreshedEventPayloadFragment on FormRefreshedEventPayload {
    id
    form {
      id
      label
      pages {
        id
        label
        groups {
          id
          label
          widgets {
            id
            __typename
            diagnostics {
              id
              kind
              message
            }
            ... on Textfield {
              label
              stringValue: value
            }
            ... on Textarea {
              label
              stringValue: value
            }
            ... on Checkbox {
              label
              booleanValue: value
            }
            ... on Select {
              label
              value
              options {
                id
                label
              }
            }
            ... on Radio {
              label
              options {
                id
                label
                selected
              }
            }
            ... on List {
              label
              items {
                id
                label
                imageURL
              }
            }
          }
        }
      }
    }
  }
`;
