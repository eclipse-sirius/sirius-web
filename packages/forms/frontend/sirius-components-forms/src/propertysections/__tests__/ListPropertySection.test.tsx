/*******************************************************************************
 * Copyright (c) 2022, 2024 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import {
  MessageOptions,
  Selection,
  SelectionContext,
  ToastContext,
  ToastContextValue,
} from '@eclipse-sirius/sirius-components-core';
import { act, cleanup, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { afterEach, expect, test, vi } from 'vitest';
import { GQLList, GQLListItem } from '../../form/FormEventFragments.types';
import { ListPropertySection, clickListItemMutation } from '../ListPropertySection';
import {
  GQLClickListItemMutationData,
  GQLClickListItemMutationVariables,
  GQLSuccessPayload,
} from '../ListPropertySection.types';

const setSelection = () => {};

crypto.randomUUID = vi.fn(() => '48be95fc-3422-45d3-b1f9-d590e847e9e1');

afterEach(() => cleanup());

const defaultListItems: GQLListItem[] = [
  {
    id: 'item1',
    label: 'item1Label',
    kind: 'itemKind',
    iconURL: [],
    deletable: true,
  },
  {
    id: 'item2',
    label: 'item2Label',
    kind: 'itemKind',
    iconURL: [],
    deletable: true,
  },
  {
    id: 'item3',
    label: 'item3Label',
    kind: 'itemKind',
    iconURL: [],
    deletable: true,
  },
];
const defaultList = {
  label: 'myList',
  iconURL: [],
  hasHelpText: false,
  items: defaultListItems,
  style: null,
  __typename: 'List',
  diagnostics: [],
  id: 'listId',
  readOnly: false,
};

const defaultListWithStyle: GQLList = {
  ...defaultList,
  style: {
    color: 'RebeccaPurple',
    bold: true,
    fontSize: 20,
    italic: true,
    strikeThrough: true,
    underline: true,
  },
};

const readOnlyList: GQLList = {
  ...defaultListWithStyle,
  readOnly: true,
};

const clickListItemVariables: GQLClickListItemMutationVariables = {
  input: {
    id: '48be95fc-3422-45d3-b1f9-d590e847e9e1',
    editingContextId: 'editingContextId',
    representationId: 'formId',
    listId: 'listId',
    listItemId: 'item3',
    clickEventKind: 'SINGLE_CLICK',
  },
};

const successPayload: GQLSuccessPayload = { messages: [], __typename: 'SuccessPayload' };
const clickListItemSuccessData: GQLClickListItemMutationData = {
  clickListItem: successPayload,
};

const mockEnqueue = vi.fn<[string, MessageOptions?], void>();

const toastContextMock: ToastContextValue = {
  enqueueSnackbar: mockEnqueue,
};

const emptySelection: Selection = {
  entries: [],
};

test('render list widget', () => {
  const { container } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectionContext.Provider value={{ selection: emptySelection, setSelection, selectedRepresentations: [] }}>
          <ListPropertySection
            editingContextId="editingContextId"
            formId="formId"
            widget={defaultListWithStyle}
            readOnly={false}
          />
        </SelectionContext.Provider>
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
});

test('render list widget with style', () => {
  const { container } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectionContext.Provider value={{ selection: emptySelection, setSelection, selectedRepresentations: [] }}>
          <ListPropertySection
            editingContextId="editingContextId"
            formId="formId"
            widget={defaultListWithStyle}
            readOnly={false}
          />
        </SelectionContext.Provider>
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
});

test('should the click event sent on item click', async () => {
  let clickListItemMutationCalled = false;
  const itemClickCalledCalledSuccessMock: MockedResponse<Record<string, any>> = {
    request: {
      query: clickListItemMutation,
      variables: clickListItemVariables,
    },
    result: () => {
      clickListItemMutationCalled = true;
      return { data: clickListItemSuccessData };
    },
  };

  const { container } = render(
    <MockedProvider mocks={[itemClickCalledCalledSuccessMock]}>
      <ToastContext.Provider value={toastContextMock}>
        <SelectionContext.Provider value={{ selection: emptySelection, setSelection, selectedRepresentations: [] }}>
          <ListPropertySection
            editingContextId="editingContextId"
            formId="formId"
            widget={defaultListWithStyle}
            readOnly={false}
          />
        </SelectionContext.Provider>
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
  const element: HTMLElement = screen.getByTestId('representation-item3Label');
  expect(element.textContent).toBe('item3Label');

  await act(async () => {
    userEvent.click(element);
    await new Promise((resolve) => setTimeout(resolve, 0));

    await waitFor(() => {
      expect(clickListItemMutationCalled).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});

test('render list widget with help hint', () => {
  const { container } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectionContext.Provider value={{ selection: emptySelection, setSelection, selectedRepresentations: [] }}>
          <ListPropertySection
            editingContextId="editingContextId"
            formId="formId"
            widget={{ ...defaultListWithStyle, hasHelpText: true }}
            readOnly={false}
          />
        </SelectionContext.Provider>
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
});

test('should render a readOnly list from widget properties', () => {
  const { container } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectionContext.Provider value={{ selection: emptySelection, setSelection, selectedRepresentations: [] }}>
          <ListPropertySection
            editingContextId="editingContextId"
            formId="formId"
            widget={readOnlyList}
            readOnly={false}
          />
        </SelectionContext.Provider>
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
});
