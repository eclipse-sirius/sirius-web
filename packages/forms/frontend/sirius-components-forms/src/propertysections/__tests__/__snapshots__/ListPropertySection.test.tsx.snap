// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render list widget 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <div
      class="makeStyles-propertySectionLabel-5"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-8 MuiTypography-subtitle2"
      >
        myList
      </h6>
    </div>
    <table
      class="MuiTable-root"
      data-testid="table-myList"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-1 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-2 makeStyles-style-3 makeStyles-style-4 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item1Label"
            >
              item1Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item1Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-1 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-2 makeStyles-style-3 makeStyles-style-4 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item2Label"
            >
              item2Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item2Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-1 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-2 makeStyles-style-3 makeStyles-style-4 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item3Label"
            >
              item3Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item3Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <p
      class="MuiFormHelperText-root"
    />
  </div>
</div>
`;

exports[`render list widget with help hint 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <div
      class="makeStyles-propertySectionLabel-29"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-32 MuiTypography-subtitle2"
      >
        myList
      </h6>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
        focusable="false"
        style="margin-left: 8px; font-size: 16px;"
        title=""
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
        />
      </svg>
    </div>
    <table
      class="MuiTable-root"
      data-testid="table-myList"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-25 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-26 makeStyles-style-27 makeStyles-style-28 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item1Label"
            >
              item1Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item1Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-25 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-26 makeStyles-style-27 makeStyles-style-28 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item2Label"
            >
              item2Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item2Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-25 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-26 makeStyles-style-27 makeStyles-style-28 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item3Label"
            >
              item3Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item3Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <p
      class="MuiFormHelperText-root"
    />
  </div>
</div>
`;

exports[`render list widget with style 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <div
      class="makeStyles-propertySectionLabel-13"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-16 MuiTypography-subtitle2"
      >
        myList
      </h6>
    </div>
    <table
      class="MuiTable-root"
      data-testid="table-myList"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-9 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-10 makeStyles-style-11 makeStyles-style-12 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item1Label"
            >
              item1Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item1Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-9 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-10 makeStyles-style-11 makeStyles-style-12 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item2Label"
            >
              item2Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item2Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-9 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-10 makeStyles-style-11 makeStyles-style-12 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item3Label"
            >
              item3Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item3Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <p
      class="MuiFormHelperText-root"
    />
  </div>
</div>
`;

exports[`should render a readOnly list from widget properties 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <div
      class="makeStyles-propertySectionLabel-37"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-40 MuiTypography-subtitle2"
      >
        myList
      </h6>
    </div>
    <table
      class="MuiTable-root"
      data-testid="table-myList"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-33 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root  makeStyles-style-35 makeStyles-style-36 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item1Label"
            >
              item1Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
              data-testid="delete-representation-item1Label"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-33 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root  makeStyles-style-35 makeStyles-style-36 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item2Label"
            >
              item2Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
              data-testid="delete-representation-item2Label"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-33 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root  makeStyles-style-35 makeStyles-style-36 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item3Label"
            >
              item3Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
              data-testid="delete-representation-item3Label"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <p
      class="MuiFormHelperText-root"
    />
  </div>
</div>
`;

exports[`should the click event sent on item click 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <div
      class="makeStyles-propertySectionLabel-21"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-24 MuiTypography-subtitle2"
      >
        myList
      </h6>
    </div>
    <table
      class="MuiTable-root"
      data-testid="table-myList"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-17 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-18 makeStyles-style-19 makeStyles-style-20 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item1Label"
            >
              item1Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item1Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-17 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-18 makeStyles-style-19 makeStyles-style-20 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item2Label"
            >
              item2Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item2Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-17 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-18 makeStyles-style-19 makeStyles-style-20 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item3Label"
            >
              item3Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item3Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <p
      class="MuiFormHelperText-root"
    />
  </div>
</div>
`;

exports[`should the click event sent on item click 2`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <div
      class="makeStyles-propertySectionLabel-21"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-24 MuiTypography-subtitle2"
      >
        myList
      </h6>
    </div>
    <table
      class="MuiTable-root"
      data-testid="table-myList"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-17 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-18 makeStyles-style-19 makeStyles-style-20 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item1Label"
            >
              item1Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item1Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-17 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-18 makeStyles-style-19 makeStyles-style-20 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item2Label"
            >
              item2Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item2Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-cell-17 MuiTableCell-sizeSmall"
          >
            <p
              class="MuiTypography-root makeStyles-canBeSelectedItem-18 makeStyles-style-19 makeStyles-style-20 MuiTypography-body1 MuiTypography-colorTextPrimary"
              data-testid="representation-item3Label"
            >
              item3Label
            </p>
            <button
              aria-label="deleteListItem"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              data-testid="delete-representation-item3Label"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <p
      class="MuiFormHelperText-root"
    />
  </div>
</div>
`;
